src/display/editor/draw.js

```diff
@@ -639,32 +649,78 @@ class DrawingEditor extends AnnotationEditor {
     unreachable("Not implemented");
   }
 
-  static startDrawing(
-    parent,
-    uiManager,
-    _isLTR,
-    { target, offsetX: x, offsetY: y }
-  ) {
+  static startDrawing(parent, uiManager, _isLTR, event) {
+    // The _currentPointerType is set when the user starts an empty drawing
+    // session. If, in the same drawing session, the user starts using a
+    // different type of pointer (e.g. a pen and then a finger), we just return.
+    //
+    // The _currentPointerId  and _currentPointerIds are used to keep track of
+    // the pointers with a same type (e.g. two fingers). If the user starts to
+    // draw with a finger and then uses a second finger, we just stop the
+    // current drawing and let the user zoom the document.
+
+    const { target, offsetX: x, offsetY: y, pointerId, pointerType } = event;
+    if (this._currentPointerType && this._currentPointerType !== pointerType) {
+      return;
+    }
+
     const {
       viewport: { rotation },
     } = parent;
     const { width: parentWidth, height: parentHeight } =
       target.getBoundingClientRect();
-    const ac = new AbortController();
+
+    const ac = (this._currentDrawingAC = new AbortController());
     const signal = parent.combinedSignal(ac);
 
+    this._currentPointerId ||= pointerId;
+    this._currentPointerType ||= pointerType;
```

Comment:
Given that `pointerId` seems to be specified as [a number](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerId#value), it might be zero?
Hence it seems overall safer just change these as follows, to prevent them being changed.
```suggestion
    this._currentPointerId ??= pointerId;
    this._currentPointerType ??= pointerType;
```